src/common/wrap/wrap-function.js

```diff
@@ -93,17 +94,28 @@ export function createWrapperWithEmitter (emitter, always) {
       // Warning: start events may mutate args!
       safeEmit(prefix + 'start', [args, originalThis, methodName], ctx, bubble)
 
+      const fnStartTime = performance.now()
       try {
         result = fn.apply(originalThis, args)
         return result
       } catch (err) {
         safeEmit(prefix + 'err', [args, originalThis, err], ctx, bubble)
-
         // rethrow error so we don't effect execution by observing.
-        throw err
+        thrownError = err
+        throw thrownError
       } finally {
-        // happens no matter what.
-        safeEmit(prefix + 'end', [args, originalThis, result], ctx, bubble)
+        const duration = performance.now() - fnStartTime
```

Comment:
I think duration needs to move up further. (And move the return out of try-catch-finally.)
like
```js
 result = fn.apply(...)
 fnEndTime = performance.now()
} catch (err) {
 fnEndTime = performance.now()
 safeEmit(...) // this can add to the overall time if included
 thrownError = err
 throw thrownError
} finally {
 const duration = fnEndTime - fnStartTime
 ...
}
return result // better practice
```